Tools/ardupilotwaf/cxx_checks.py

```diff
@@ -173,3 +173,11 @@ def check_libiio(cfg, env):
     if cfg.env['LIB_LIBIIO']:
         env.LIB += ['dl']
     return True
+
+@conf
+def check_libdl(cfg, env):
+    ret = cfg.check(compiler='cxx', lib='dl')
+    if ret:
+        env.LIB += cfg.env['LIB_DL']
+        cfg.define('HAVE_LIBDL', 1)
+    return ret
```

Comment:
Please look at the other lib detection functions, we should keep it as consistent as possible - and I'm almost sure the way it is now makes libdl mandatory.